File: pkg/codegen/testing/test/testdata/aws-s3-logging-pp/nodejs/aws-s3-logging.ts
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";

const logs = new aws.s3.Bucket("logs", {});
const bucket = new aws.s3.Bucket("bucket", {loggings: [{
    targetBucket: logs.bucket,
}]});
export const targetBucket = bucket.loggings.apply(loggings => loggings?[0]?.targetBucket);

File: pkg/codegen/testing/utils/testdata/aws-s3-logging-pp/dotnet/aws-s3-logging.cs
using System.Collections.Generic;
using System.Linq;
using Pulumi;
using Aws = Pulumi.Aws;

return await Deployment.RunAsync(() => 
{
    var logs = new Aws.S3.Bucket("logs");

    var bucket = new Aws.S3.Bucket("bucket", new()
    {
        Loggings = new[]
        {
            new Aws.S3.Inputs.BucketLoggingArgs
            {
                TargetBucket = logs.BucketName,
            },
        },
    });

    return new Dictionary<string, object?>
    {
        ["targetBucket"] = bucket.Loggings.Apply(loggings => loggings[0]?.TargetBucket),
    };
});

